util/create-brackets-table.raku

```diff
@@ -1,55 +1,65 @@
 #!/usr/bin/env raku
 
-# global array defined in the BEGIN block at the end:
-my @bracket-chars;
+# TODO
+#   + get desired exit behavior from Coke
+#   + settle on table name and output file name
 
-my $f = $*PROGRAM.parent(2).child('doc').child('Language').child('brackets.pod6');
 
 if !@*ARGS {
     say qq:to/HERE/;
-    Usage: {$*PROGRAM.IO.basename} go | reorder
+    Usage: {$*PROGRAM.IO.basename} go [refresh][debug]
 
     Writes the HLL::Grammar '\$brackets' chars into a Pod6 table.
 
-    If the 'reorder' mode is chosen, the output table is sorted
-    by the value of the codepoint of the opening character.
-    Otherwise, the order of the source definition is used.
+    The HLL::Grammar.nqp file as source is updated if it
+    is not found in /util or the 'refresh' option is used.
+
     HERE
     exit;
 }
 
-my $reorder = 0;
-my $arg = @*ARGS.shift;
-if $arg ~~ /:i r/ {
-    ++$reorder;
+my $refresh       = 0;
+my $want-original = 0;
+my $debug         = 0;
+
+for @*ARGS {
+    when /^ r/ { ++$refresh }
+    when /^ d/ { ++$debug }
 }
-say "Reorder = $reorder";
 
-my @ofils;
+# The output table file:
+my $repopath = $*PROGRAM.IO.absolute.IO.parent(2);
+my $fdir1 = $repopath ~ "/doc/Language";
+my $f1    = "$fdir1/brackets.pod6";
 
-my $lb = '|';
-my $rb = '|';
+# The local copy of HLL::Grammar.nqp:
+my $fdir2 = $repopath ~ "/util";
+my $f2    = "$fdir2/Grammar.nqp";
 
-if $reorder {
-    $f = "../doc/Language/brackets-reordered.pod6";
+if $debug {
+    say "DEBUG: paths:";
+    say "  output table: $f1";
+    say "  Grammar.nqp : $f2";
+    say "DEBUG exit"; exit;
 }
 
-if $lb ~~ /'|'/ {
-    # need to escape the pipe for the doc site
-    $lb = '\\|';
-    $rb = $lb
-}
-write-brackets-pod6-file :$f, :$lb, :$rb, :$reorder;
-@ofils.append: $f;
+# This array is defined dynamically by reading the source
+# file at https://github.com/Raku/nqp/src/HLL/Grammar.nqp.
```

Comment:
This says it's coming from github, but the file is opened locally. We might want to emulate RAKUDO_SRC env var used elsewhere in the xt/ tests, and use NQP_SRC to find the local nqp checkout.

Reply:
So done.